apps/total-typescript/src/components/lesson-list.tsx

```diff
@@ -239,16 +239,18 @@ const SolutionLink = ({
 }: {
   module: SanityDocument
   section?: SanityDocument
-  exercise: Exercise
+  exercise: LessonResource
   path: string
 }) => {
   const router = useRouter()
-  const solution = exercise.solution
+  const {data: solution, status} = trpc.solutions.getSolution.useQuery({
+    exerciseSlug: router.query.exercise as string,
+  })
```

Comment:
a `LessonResource` doesn't specifically have a `solution` (an Exercise is a superset) so we load them if we can async